_posts/2021-12-15-dd_00004_logic_gates_2.md

```diff
@@ -0,0 +1,343 @@
+  <img src="/assets/images/dd-00003-excercise.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+A little warning: don't be scared, this is the transistor implementation of the
+same circuit:
+
+<figure class="img-center" style="width:98%">
+  <img src="/assets/images/dd-00004-complex.jpg" alt="complex CMOS circuit">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+To make it easier for you, we have circled the logic gates in different colors.
+We can recognize the CMOS transistors of the AND, OR and NAND gates. The CMOS
+XOR we didn't show you earlier, but from the circuit you can see that it looks
+much more complex than the other gates. Overall, this circuit made up of just
+four logic gates already consumes 28 individual transistors! Drawing this by
+hand is still somewhat manageble. But think of a big processor: those beasts
+consist of _millions_ of logic gates (_not_ transistors). And since each gate is
+made up of multiple transistors itself, you can imagine, that designing an
+entire CPU by hand is straight up infeasible nowadays! That's why we have to use
+the help of **Electronic Design Automation** (EDA), i.e. software to automate
+the design of digital circuits, when we build such complex chips. These tools
+will do the mapping from logic gates to transistors fully automatic! If you are
+interested, we can do a post about EDA, what tools there are and how they work;
+let us know in the comments (we might do it anyways ;)
+
+## Physical Layout
+
+One final note. Remember when we said in our last post, that the symbols we use
+for drawing logic gates were not how the actual gate would look in hardware?
+Well, even the transistor diagrams we showed you in this post don't actually
+depict how the real hardware looks, but they are already pretty close! In case
+you are wondering, here is how a NAND _actually_ looks on the final chip:
+
+<figure class="img-center" style="width:30%">
+  <img src="/assets/images/dd-00004-cmos-nand-layout.svg" alt="CMOS NAND Layout">
+  <figcaption class="text-center">
+   (source: Wikipedia)
+  </figcaption>
+</figure>
+
+"This looks nothing like the schematics you showed us in this post!?", you may
+ask yourself. That is a totally valid question! Let us explain what's going on
+here. If you have read our very [first post](/categories/digital_design/intro)
+in the Digital Design series, you may remember that there are a lot of steps
+involved when making a microchip. One of these steps is **physical design**,
+which comes shortly before manufacturing the chip. In this step, the circuit
+design consisting of thousands or millions of logic gates is transformed into a
+so-called **layout**. The layout describes how the individial transistors and
+the wires connecting them should be _layed out_ on the final chip. The layout is
+the final _blueprint_ for the manufacturing. We have to admit that the picture
```

Comment:
Maybe you can mention somehwere here that the layout is also the step where physical aspects as the size, heat and so on are considered

Reply:
Like this? 5926128

Reply:
maybe revove the "also" in note two? The additionally has already the meaning. But if you want you can leave it

Reply:
fixed 173087d